rust/src/main.rs

```diff
@@ -32,9 +48,16 @@ fn setup_logging() -> Result<(), fern::InitError> {
             _ => log::LevelFilter::Info,
         },
         // Default to info
-        Err(_) => log::LevelFilter::Info
+        Err(_) => log::LevelFilter::Info,
     };
 
+    let logging_directory = get_logging_directory()?;
```

Comment:
I might call this function `path_for_log_file(filename: &str)` (or, better,` path_for_log_file<P: AsRef<Path>>(filename: P)`), and pass it in the name of the file; and then I would move in the `create_dir_all` and the path appending to that function. This would make it easier to eventually share this with another binary.